plugins/eddn.py

```diff
@@ -633,6 +631,277 @@ def export_journal_entry(self, cmdr: str, is_beta: bool, entry: Mapping[str, Any
+
+    def export_journal_navbeaconscan(
+            self, cmdr: str, system_name: str, is_beta: bool, entry: Mapping[str, Any]
+    ) -> Optional[str]:
+        """
+        Send an NavBeaconScan to EDDN on the correct schema.
+
+        :param cmdr: the commander under which this upload is made
+        :param system_name: Name of the current system.
+        :param is_beta: whether or not we are in beta mode
+        :param entry: the journal entry to send
+        """
+        # { "timestamp":"2021-09-24T13:57:24Z", "event":"NavBeaconScan", "SystemAddress":670417626481, "NumBodies":18 }
+        #######################################################################
+        # Elisions
+        entry = filter_localised(entry)
+        #######################################################################
+
+        #######################################################################
+        # Augmentations
+        #######################################################################
+        ret = this.eddn.entry_augment_system_data(entry, system_name)
+        if isinstance(ret, str):
+            return ret
+
+        entry = ret
+        #######################################################################
+
+        msg = {
+            '$schemaRef': f'https://eddn.edcd.io/schemas/navbeaconscan/1{"/test" if is_beta else ""}',
+            'message': entry
+        }
+
+        this.eddn.export_journal_entry(cmdr, entry, msg)
+        return None
+
+    def export_journal_codexentry(
+            self, cmdr: str, is_beta: bool, entry: Mapping[str, Any]
+    ) -> Optional[str]:
+        """
+        Send a CodexEntry to EDDN on the correct schema.
+
+        :param cmdr: the commander under which this upload is made
+        :param is_beta: whether or not we are in beta mode
+        :param entry: the journal entry to send
+        """
+        # {
+        #   "timestamp":"2021-09-26T12:29:39Z",
+        #   "event":"CodexEntry",
+        #   "EntryID":1400414,
+        #   "Name":"$Codex_Ent_Gas_Vents_SilicateVapourGeysers_Name;",
+        #   "Name_Localised":"Silicate Vapour Gas Vent",
+        #   "SubCategory":"$Codex_SubCategory_Geology_and_Anomalies;",
+        #   "SubCategory_Localised":"Geology and anomalies",
+        #   "Category":"$Codex_Category_Biology;",
+        #   "Category_Localised":"Biological and Geological",
+        #   "Region":"$Codex_RegionName_18;",
+        #   "Region_Localised":"Inner Orion Spur",
+        #   "System":"Bestia",
+        #   "SystemAddress":147916327267,
+        #   "Latitude":23.197777, "Longitude":51.803349,
+        #   "IsNewEntry":true,
+        #   "VoucherAmount":50000
+        # }
+        #######################################################################
+        # Elisions
+        entry = filter_localised(entry)
+        # Keys specific to this event
+        for k in ('IsNewEntry', 'NewTraitsDiscovered'):
+            if k in entry:
+                del entry[k]
```

Comment:
This could be done with a  pop, not sure if we want to?

Reply:
I wonder if that was another victim of Mapping instead of MutableMapping, because it's what I tried initially and only got an error.

Reply:
Very likely, `MutableMapping` has all of the pop etc stuff on it

Reply:
It throws KeyError if they don't exist, which they might not in this case, so I'll leave it as-is.  `del` versus `pop` isn't really going to change anything without the `in` check.